govuk-notify-spi/src/main/scala/uk/gov/nationalarchives/notifyspi/NotifyEmailSenderProvider.scala

```diff
@@ -1,30 +1,43 @@
 package uk.gov.nationalarchives.notifyspi
 
 import java.util
-
 import org.keycloak.email.{EmailException, EmailSenderProvider}
 import org.keycloak.models.UserModel
+import software.amazon.awssdk.http.apache.ApacheHttpClient
+import software.amazon.awssdk.regions.Region
 import uk.gov.service.notify.NotificationClient
 
+import java.net.URI
 import scala.jdk.CollectionConverters._
 import scala.util.{Failure, Success, Try}
+import software.amazon.awssdk.services.ssm.SsmClient
+import software.amazon.awssdk.services.ssm.model.GetParameterRequest
+import com.typesafe.config.{ConfigFactory, Config => TypeSafeConfig}
 
 class NotifyEmailSenderProvider(environmentVariables: Map[String, String]) extends EmailSenderProvider {
+  val configFactory: TypeSafeConfig = ConfigFactory.load
+  val ssmEndpoint: String = configFactory.getString("ssm.endpoint")
+  val notifyApiKeyPath: String = configFactory.getString("notify.apiKeyPath")
+  val notifyTemplateIdPath: String = configFactory.getString("notify.templateIdPath")
 
   private implicit class EnvironmentalVariablesUtils(map: Map[String, String]) {
     def getApiKey: String = {
-      retrieveValue("GOVUK_NOTIFY_API_KEY")
+      getSsmParameterValue(notifyApiKeyPath)
     }
 
     def getTemplateId: String = {
-      retrieveValue("GOVUK_NOTIFY_TEMPLATE_ID")
+      getSsmParameterValue(notifyTemplateIdPath)
     }
 
-    private def retrieveValue(key: String): String = {
-      map.get(key) match {
-        case Some(value) => value
-        case _ => throw new EmailException(s"Missing '$key' value")
-      }
+    private def getSsmParameterValue(parameterPath: String): String = {
+      val httpClient = ApacheHttpClient.builder.build
+      val ssmClient: SsmClient = SsmClient.builder()
+        .endpointOverride(URI.create(ssmEndpoint))
```

Comment:
The only reason to do endpoint override is if you're testing the SSM call with wiremock because you can override it in the test application.conf. It doesn't look like you're adding tests for the call to SSM to you should be able to get rid of this line as it has a sensible default.